extensions/cornerstone/src/utils/interleave.test.ts

```diff
@@ -0,0 +1,304 @@
+
+    expect(result).toEqual([1, 2, 3, 4, 5, 6]);
+  });
+
+  it('should interleave three arrays of equal length', () => {
+    const input = [
+      [1, 4, 7],
+      [2, 5, 8],
+      [3, 6, 9],
+    ];
+
+    const result = interleave(input);
+
+    expect(result).toEqual([1, 2, 3, 4, 5, 6, 7, 8, 9]);
+  });
+
+  it('should handle arrays of different lengths', () => {
+    const input = [
+      [1, 4],
+      [2, 5, 7, 8],
+      [3, 6],
+    ];
+
+    const result = interleave(input);
+
+    expect(result).toEqual([1, 2, 3, 4, 5, 6, 7, 8]);
+  });
+
+  it('should handle first array being longer', () => {
+    const input = [
+      [1, 3, 5, 7, 9, 11],
+      [2, 4, 6],
+    ];
+
+    const result = interleave(input);
+
+    expect(result).toEqual([1, 2, 3, 4, 5, 6, 7, 9, 11]);
+  });
+
+  it('should handle second array being longer', () => {
+    const input = [
+      [1, 3, 5],
+      [2, 4, 6, 8, 10, 12],
+    ];
+
+    const result = interleave(input);
+
+    expect(result).toEqual([1, 2, 3, 4, 5, 6, 8, 10, 12]);
+  });
+
+  it('should handle empty arrays in the input', () => {
+    const input = [[1, 3, 5], [], [2, 4, 6]];
+
+    const result = interleave(input);
+
+    expect(result).toEqual([1, 2, 3, 4, 5, 6]);
+  });
+
+  it('should handle all empty arrays', () => {
+    const input = [[], [], []];
+
+    const result = interleave(input);
+
+    expect(result).toEqual([]);
+  });
+
+  it('should handle single element arrays', () => {
+    const input = [[1], [2], [3], [4]];
+
+    const result = interleave(input);
+
+    expect(result).toEqual([1, 2, 3, 4]);
+  });
+
+  it('should handle mixed single and multi-element arrays', () => {
+    const input = [[1], [2, 5, 8], [3], [4, 6, 7, 9]];
+
+    const result = interleave(input);
+
+    expect(result).toEqual([1, 2, 3, 4, 5, 6, 8, 7, 9]);
+  });
+
+  it('should handle string arrays', () => {
+    const input = [
+      ['a', 'c', 'e'],
+      ['b', 'd', 'f'],
+    ];
+
+    const result = interleave(input);
+
+    expect(result).toEqual(['a', 'b', 'c', 'd', 'e', 'f']);
+  });
+
+  it('should handle object arrays', () => {
+    const obj1 = { id: 1 };
+    const obj2 = { id: 2 };
+    const obj3 = { id: 3 };
+    const obj4 = { id: 4 };
+
+    const input = [
+      [obj1, obj3],
+      [obj2, obj4],
+    ];
+
+    const result = interleave(input);
+
+    expect(result).toEqual([obj1, obj2, obj3, obj4]);
+  });
+
+  it('should handle large arrays efficiently', () => {
+    const array1 = Array.from({ length: 1000 }, (_, i) => i * 2);
+    const array2 = Array.from({ length: 1000 }, (_, i) => i * 2 + 1);
+    const input = [array1, array2];
+
+    const startTime = performance.now();
+    const result = interleave(input);
+    const endTime = performance.now();
+
+    expect(result).toHaveLength(2000);
+    expect(result[0]).toBe(0);
+    expect(result[1]).toBe(1);
+    expect(result[2]).toBe(2);
+    expect(result[3]).toBe(3);
+    expect(endTime - startTime).toBeLessThan(100);
```

Comment:
This one bothers me a little. I would assume the actual time is MUCH less than 100 right? If not, then I would consider choosing something a little higher.